framework/audio/engine/internal/audioplayer.cpp

```diff
@@ -81,13 +96,13 @@ void AudioPlayer::pause()
 void AudioPlayer::stop()
 {
     doStop();
-    m_status.set(PlayStatus::STOPED);
+    //! NOTE The status will be changed in `onStop`
```

Comment:
Why we're setting this status for "pause" directly in "void AudioPlayer::pause()"?
It's a bit confusing

Reply:
Because the stop status can be set in two ways when the player is called to stop, and when the audioengine it destroys the sound source because it has ended.

See https://github.com/musescore/MuseScore/pull/6371#discussion_r460394460

Reply:
I do wonder if we should keep the audio source after its end, because you can just replay the same source again. So maybe it's a good point to destroy the source only when we're going to play another source.

Reply:
This is `SoLoud` behaviour.